Cabal/Distribution/PackageDescription/Check.hs

```diff
@@ -1031,6 +1031,8 @@ checkPaths pkg =
       ++ concat
          [    [ (path, "c-sources")        | path <- cSources        bi ]
            ++ [ (path, "cxx-sources")      | path <- cxxSources      bi ]
+           ++ [ (path, "s-sources")        | path <- sSources        bi ]
```

Comment:
IMO `asm-sources` would be a better name. One question is whether we should have `asm-x86-sources`, `asm-x86_64-sources`, etc. instead of a single catch-all field... probably not.

Reply:
I'd argue for using `if arch(...):` in those cases.

I'm fine with `asm-soures` as well.

Reply:
Yep, I also think that `if(arch)` is the right approach.

Reply:
especially given @phadej's `elif` using `if arch(...)` is quite ok I think